features/steps/steps.go

```diff
@@ -39,17 +44,76 @@ func (c *PopulationTypesComponent) cantabularIsUnresponsive() error {
 	return nil
 }
 
-func (c *PopulationTypesComponent) theServiceRespondsWithAnInternalServerErrorSaying(expected string) error {
+func (c *PopulationTypesComponent) theServiceRespondsWithAnInternalServerErrorSaying(expectedHttpCode int, expected string) error {
 	resp := c.apiFeature.HttpResponse
-	if resp.StatusCode != http.StatusInternalServerError {
+	if resp.StatusCode != expectedHttpCode {
 		return fmt.Errorf("expected: %d. actual: %d", http.StatusInternalServerError, resp.StatusCode)
 	}
 	body, err := ioutil.ReadAll(c.apiFeature.HttpResponse.Body)
 	if err != nil {
 		return err
 	}
-	if !strings.Contains(string(body), fakeCantabularFailedToRespondErrorMessage) {
+	if !strings.Contains(string(body), expected) {
 		return fmt.Errorf("expected to contain: %s. actual: %s", fakeCantabularFailedToRespondErrorMessage, string(body))
 	}
 	return nil
 }
+
+// theFollowingCantabularResponseIsAvailable generates a mocked response for Cantabular Server POST /graphql
+func (c *PopulationTypesComponent) theFollowingCantabularResponseIsAvailable() error {
+	gd := &cantabular.GetGeographyDimensionsResponse{Dataset: gql.DatasetRuleBase{RuleBase: gql.RuleBase{
+		IsSourceOf: gql.Variables{Edges: []gql.Edge{
+			{
+				Node: gql.Node{
+					Name:       "country",
+					Label:      "Country",
+					Categories: gql.Categories{TotalCount: 2},
+					MapFrom: []gql.Variables{{Edges: []gql.Edge{{
+						Node: gql.Node{
+							Name:       "city",
+							Label:      "City",
+							FilterOnly: "false",
+						},
+					}}}},
+				},
+			},
+			{
+				Node: gql.Node{
+					Name:       "city",
+					Label:      "City",
+					Categories: gql.Categories{TotalCount: 3},
+					MapFrom:    []gql.Variables{},
+				},
+			},
+		}},
+	}}}
+	c.fakeCantabularGeoDimensions = gd
+	return nil
+}
+
+func (c *PopulationTypesComponent) anErrorJsonResponseIsReturnedFromCantabularApiExtension() error {
+	c.fakeCantabularGeoDimensions = nil
+	c.fakeCantabularIsUnresponsive = true
+	return nil
+}
+
+func (c *PopulationTypesComponent) aListOfAreaTypesIsReturned() error {
+	return c.apiFeature.IShouldReceiveTheFollowingJSONResponse(&godog.DocString{
```

Comment:
Would it be better to keep the (potentially) dynamic data to the feature file? I think it's clearer to be able to see the expected inputs/outputs all in that one file.

Reply:
The style godog tests in this project were already set in this way.
Probably something to refactor the next time around, if we really want to align with other repos.